risc0/cargo-risczero/src/commands/build_guest.rs

```diff
@@ -0,0 +1,131 @@
+use crate::utils::{ensure_binary, CommandExt};
+
+const DOCKER_IGNORE: &'static str = "**/elfs\n**/target\n**/Dockerfile\n**/.git";
+/// `cargo risczero build-guest`
+#[derive(Parser)]
+pub struct BuildGuest {
+    /// Location of the Cargo.toml of the guest code
+    ///
+    /// This path is relative to the current directory
+    #[clap(value_parser, long)]
+    pub manifest_path: String,
+}
+
+impl BuildGuest {
+    pub fn run(&self) -> Result<()> {
+        let meta = MetadataCommand::new()
+            .manifest_path(self.manifest_path.as_str())
+            .exec()?;
+        let pkg_name = &meta
+            .root_package()
+            .context("failed to parse Cargo.toml")?
+            .name;
+        eprintln!("Building the riscv32im-risc0-zkvm-elf binary for {pkg_name}...");
+        ensure_binary("docker", &["--version"])?;
+        let package_name = pkg_name.replace('-', "_");
+        self.create_dockerfile(package_name.as_str())?;
+        self.build()?;
+
+        let paths = fs::read_dir(&format!("./elfs/{package_name}/"))?;
+        eprintln!("ELFs ready at:");
+        for path in paths {
+            let entry = path.unwrap().path();
+            eprintln!("{} - ImageID: {}", entry.display(), self.image_id(&entry)?);
+        }
+
+        Ok(())
+    }
+
+    /// Create the dockerfile.
+    ///
+    /// Overwrites if a dockerfile already exists.
+    fn create_dockerfile(&self, pkg_name: &str) -> Result<()> {
+        let manifest_env = &[("CARGO_MANIFEST_PATH", self.manifest_path.as_str())];
+        let pkg_env = &[("PKG_NAME", pkg_name)];
+        let c_flags = format!(
+            "-C passes=loweratomic -C link-arg=-Ttext=0x{:08X} -C link-arg=--fatal-warnings",
+            memory::TEXT_START
+        );
+        let c_flags_env = &[("RUSTFLAGS", c_flags.as_str())];
+
+        let build = DockerFile::new()
+            .from_alias("build", "risczero/risc0-guest-builder:v0.17")
+            .workdir("/src")
+            .copy(".", ".")
+            .env(manifest_env)
+            .env(c_flags_env)
+            .run(
+                "cargo +risc0 fetch --target riscv32im-risc0-zkvm-elf --manifest-path $CARGO_MANIFEST_PATH --locked",
+            )
+            .run(
+                "CARGO_TARGET_DIR=target \\\n\
+                \tcargo +risc0 build \\\n\
+                \t--locked \\\n\
+                \t--release \\\n\
+                \t--target riscv32im-risc0-zkvm-elf \\\n\
+                \t--manifest-path $CARGO_MANIFEST_PATH")
+            .run(r"find target/riscv32im-risc0-zkvm-elf/release -maxdepth 1 -type f -exec test -x {} \; -exec cp {} /tmp/ \;");
+
+        let binary: DockerFile<'_> = DockerFile::new()
+            .comment("binary stage")
+            .from_alias("binary", "scratch")
+            .env(pkg_env)
+            .copy_from("build", "/tmp", "/$PKG_NAME");
+
+        let file = DockerFile::new().dockerfile(build).dockerfile(binary);
+        let mut dockerfile = File::create("Dockerfile")?;
```

Comment:
Do we cleanup these temp files anywhere?

Reply:
Added 👍